label_studio/ml_models/models.py

```diff
@@ -60,6 +63,16 @@ def full_title(self):
 
     prompt = models.TextField(_('prompt'), null=False, blank=False, help_text='Prompt to execute')
 
+    def delete_predictions(self):
+        """
+        Deletes any predictions that have originated from a ModelVersion
+
+        Currently assumes that we are setting the `model_version` field of Prediction
+        to ModelVersion.full_title
```

Comment:
I don't think this is true. Very confusingly, `Prediction.model_version` was added during the recent MLBackend changes, it describes the version of an MLBackend, and is entirely unrelated to this ModelVersion. We are setting `Prediction.model_run`: https://github.com/HumanSignal/label-studio/blob/fb-dia-957/label_studio/tasks/models.py#L869 

I don't think we are setting `Prediction.model_version`, and if we are, we should stop, since it'd show up in the UI for autolabeling using an MLBackend and confuse people, afaiui.

Reply:
Yeah, it's possible that it's actually ok, but that's not what the field was added for

Reply:
Yeah its not the exact use of this field, but I think we do need to set it actually. Its whats shown on the project side as what made this prediction, its just used as the name of the model essentially. 
If we dont want to reference it this way, we need to add a new foreign key for prompter ModelVersion as well, just ModelRun wont do it, since when we update a ModelVersion we want to delete existing Predictions. 
But I guess we enforce that there is only one ModelRun per ModelVersion, so could be ok. Will try it